lib/create-contentful-api.ts

```diff
@@ -710,6 +658,11 @@ export default function createContentfulApi<OptionType>(
       withAllLocales: false,
     }
   ) {
+    const { withAllLocales } = options
+
+    validateLocaleParam(query, !!withAllLocales)
```

Comment:
`withAllLocales` instead of `!!withAllLocales` maybe since it's gonna be boolean or undefined anyway

Reply:
I added the `!!` so I we can have `validateLocaleParam` take a boolean as second param (see [here](https://github.com/contentful/contentful.js/pull/1140/files#diff-32b5706a8b1c87b87bdd11d85069a594c56b3aa6459daa7b91aef7a4200d84e1R19))